BlinkID/android/src/main/java/com/microblink/flutter/MicroblinkCreationParams.kt

```diff
@@ -0,0 +1,40 @@
+package com.microblink.flutter
+
+
+interface MicroblinkCreationParamsFactory {
+    fun fromMap(args: Map<*, *>): MicroblinkCreationParams
+}
+
+interface OverlaySettingsFactory {
+    fun fromMap(args: Map<*, *>): OverlaySettings
+}
+
+
+public class MicroblinkCreationParams(
+    val overlaySettings: OverlaySettings,
+    val licenseKey: String,
+    val recognizerCollection: Map<String, *>
+) {
+    companion object : MicroblinkCreationParamsFactory {
+        override fun fromMap(args: Map<*, *>): MicroblinkCreationParams {
+            @Suppress("UNCHECKED_CAST")
+            return MicroblinkCreationParams(
+                overlaySettings = OverlaySettings.fromMap(args["overlaySettings"] as Map<*, *>),
+                licenseKey = args["licenseKey"] as String,
+                recognizerCollection = args["recognizerCollection"] as Map<String, *>
+            )
+
+        }
+    }
+}
+
+public class OverlaySettings(val useFrontCamera: Boolean = false, val flipFrontCamera: Boolean = false) {
+    companion object : OverlaySettingsFactory {
+        override fun fromMap(args: Map<*, *>): OverlaySettings {
+            return OverlaySettings(
+                useFrontCamera = (args["useFrontCamera"] as Boolean?) ?: false,
+                flipFrontCamera = (args["flipFrontCamera"] as Boolean?) ?: false
```

Comment:
I prefer to rename `flipFrontCamera` to `flipFrontCameraPreview` as were flipping only the preview. Apart from that LGTM

Reply:
Yeah, `flipFrontCamera` sounds fine, as you mentioned, whether the actual image is flipped or not, is out of user's concern.